src/Database/Relations/DeferOneOrMany.php

```diff
@@ -48,13 +51,16 @@ public function withDeferred($sessionKey)
 
         // Unbind (Remove)
         $newQuery->whereNotExists(function($query) use ($sessionKey) {
+            if(DbDongle::getDriver() == 'sqlsrv') {
+				$query = $query->select('*');
+			}
             $query->from('deferred_bindings')
                 ->whereRaw(DbDongle::cast('slave_id', 'INTEGER').' = '.DbDongle::getTablePrefix().$this->related->getQualifiedKeyName())
                 ->where('master_field', $this->relationName)
                 ->where('master_type', get_class($this->parent))
                 ->where('session_key', $sessionKey)
                 ->where('is_bind', false)
-                ->whereRaw(DbDongle::parse('id > ifnull((select max(id) from '.DbDongle::getTablePrefix().'deferred_bindings where
+                ->whereRaw(DbDongle::parse('id > i'.(DbDongle::getDriver() == 'sqlsrv' ? 's' : 'f').'null((select max(id) from '.DbDongle::getTablePrefix().'deferred_bindings where
```

Comment:
Without this change you get this:

     ERROR
    We're sorry, but an unhandled error occurred. Please see the details below.

    SQLSTATE[42000]: [Microsoft][ODBC Driver 11 for SQL Server][SQL Server]'ifnull' is not a recognized built-in function name. (SQL: select * from [card_seo] where ([card_seo].[attachment_id] = 1 and [card_seo].[attachment_id] is not null and [card_seo].[attachment_type] = ExitControl\RichIndexPageManager\Models\RichPage and [field] = seotag or exists (select * from [deferred_bindings] where slave_id = card_seo.seo_id and [master_field] = seotag and [master_type] = ExitControl\RichIndexPageManager\Models\RichPage and [session_key] = Vg5na53JCe1iSpdrgWHqExSesnyFhwrLSWNrx4ns and [is_bind] = 1)) and not exists (select * from [deferred_bindings] where slave_id = card_seo.seo_id and [master_field] = seotag and [master_type] = ExitControl\RichIndexPageManager\Models\RichPage and [session_key] = Vg5na53JCe1iSpdrgWHqExSesnyFhwrLSWNrx4ns and [is_bind] = 0 and id > ifnull((select max(id) from deferred_bindings where slave_id = card_seo.seo_id and master_field = seotag and master_type = ExitControl\RichIndexPageManager\Models\RichPage and session_key = Vg5na53JCe1iSpdrgWHqExSesnyFhwrLSWNrx4ns and is_bind = 1 ), 0)) order by [sort_order] asc)
    F:\MassWeb\nl\newyorkhotel\wp\office\vendor\laravel\framework\src\Illuminate\Database\Connection.php line 624
    TYPE

    Undefined

    EXCEPTION

    Illuminate\Database\QueryException

    618
    619
    620
    621
    622
    623
    624
    625
    626
    627
    628
    629
    630
      
            // If an exception occurs when attempting to run a query, we'll format the error
            // message to include the bindings with SQL, which will make this exception a
            // lot more helpful to the developer instead of just the database's errors.
            catch (Exception $e)
            {
                throw new QueryException(
                    $query, $this->prepareBindings($bindings), $e
                );
            }
      
            return $result;
        }
     STACK TRACE
    #	CALLED CODE	DOCUMENT	LINE
    86	Illuminate\Database\Connection->runQueryCallback(…)	~/vendor/laravel/framework/src/Illuminate/Database/Connection.php	580
    85	Illuminate\Database\Connection->run(…)	~/vendor/laravel/framework/src/Illuminate/Database/Connection.php	297
    84	Illuminate\Database\Connection->select(…)	~/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php	1377
    83	Illuminate\Database\Query\Builder->runSelect()	~/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php	1367
    82	Illuminate\Database\Query\Builder->getFresh(…)	~/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php	1354
    81	Illuminate\Database\Query\Builder->get(…)	~/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php	380
    80	Illuminate\Database\Eloquent\Builder->getModels(…)	~/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php	162
    79	Illuminate\Database\Eloquent\Builder->get()	~/plugins/exitcontrol/richindexpagemanager/formwidgets/SeoCardMaker.php	44
    78	ExitControl\RichIndexPageManager\FormWidgets\SeoCardMaker->getCard()	~/plugins/exitcontrol/richindexpagemanager/formwidgets/SeoCardMaker.php	47
    77	ExitControl\RichIndexPageManager\FormWidgets\SeoCardMaker->getSeoCard()	~/plugins/exitcontrol/richindexpagemanager/formwidgets/SeoCardMaker.php	133
    76	ExitControl\RichIndexPageManager\FormWidgets\SeoCardMaker->prepareVars()	~/plugins/exitcontrol/richindexpagemanager/formwidgets/SeoCardMaker.php	121
    75	ExitControl\RichIndexPageManager\FormWidgets\SeoCardMaker->render()	~/modules/backend/widgets/form/partials/_field_widget.htm	5
    74	include(…)	~/modules/system/traits/ViewMaker.php	208
    73	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    72	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/Form.php	247
    71	Backend\Widgets\Form->renderFieldElement(…)	~/modules/backend/widgets/form/partials/_field.htm	19
    70	include(…)	~/modules/system/traits/ViewMaker.php	208
    69	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    68	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_field-container.htm	8
    67	include(…)	~/modules/system/traits/ViewMaker.php	208
    66	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    65	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_form_fields.htm	2
    64	include(…)	~/modules/system/traits/ViewMaker.php	208
    63	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    62	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_form_tabs.htm	27
    61	include(…)	~/modules/system/traits/ViewMaker.php	208
    60	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    59	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_section.htm	27
    58	include(…)	~/modules/system/traits/ViewMaker.php	208
    57	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    56	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_form.htm	7
    55	include(…)	~/modules/system/traits/ViewMaker.php	208
    54	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    53	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/form/partials/_form-container.htm	8
    52	include(…)	~/modules/system/traits/ViewMaker.php	208
    51	Backend\Classes\WidgetBase->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	69
    50	Backend\Classes\WidgetBase->makePartial(…)	~/modules/backend/widgets/Form.php	214
    49	Backend\Widgets\Form->render(…)	~/modules/backend/behaviors/FormController.php	338
    48	Backend\Behaviors\FormController->formRender()		
    47	call_user_func_array(…)	~/vendor/october/rain/src/Extension/ExtendableTrait.php	354
    46	October\Rain\Extension\Extendable->extendableCall(…)	~/vendor/october/rain/src/Extension/Extendable.php	44
    45	October\Rain\Extension\Extendable->__call(…)	~/plugins/exitcontrol/richindexpagemanager/controllers/richpage/update.htm	13
    44	ExitControl\RichIndexPageManager\Controllers\RichPage->formRender()	~/plugins/exitcontrol/richindexpagemanager/controllers/richpage/update.htm	13
    43	include(…)	~/modules/system/traits/ViewMaker.php	208
    42	Backend\Classes\Controller->makeFileContents(…)	~/modules/system/traits/ViewMaker.php	81
    41	Backend\Classes\Controller->makeView(…)	~/modules/backend/classes/Controller.php	369
    40	Backend\Classes\Controller->execPageAction(…)	~/modules/backend/classes/Controller.php	251
    39	Backend\Classes\Controller->run(…)	~/modules/backend/classes/BackendController.php	95
    38	Backend\Classes\BackendController->run(…)		
    37	call_user_func_array(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Controller.php	246
    36	Illuminate\Routing\Controller->callAction(…)	~/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php	162
    35	Illuminate\Routing\ControllerDispatcher->call(…)	~/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php	107
    34	Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(…)		
    33	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	141
    32	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)		
    31	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	101
    30	Illuminate\Pipeline\Pipeline->then(…)	~/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php	108
    29	Illuminate\Routing\ControllerDispatcher->callWithinStack(…)	~/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php	67
    28	Illuminate\Routing\ControllerDispatcher->dispatch(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Route.php	204
    27	Illuminate\Routing\Route->runWithCustomDispatcher(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Route.php	134
    26	Illuminate\Routing\Route->run(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Router.php	701
    25	Illuminate\Routing\Router->Illuminate\Routing\{closure}(…)		
    24	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	141
    23	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)		
    22	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	101
    21	Illuminate\Pipeline\Pipeline->then(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Router.php	703
    20	Illuminate\Routing\Router->runRouteWithinStack(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Router.php	670
    19	Illuminate\Routing\Router->dispatchToRoute(…)	~/vendor/laravel/framework/src/Illuminate/Routing/Router.php	628
    18	Illuminate\Routing\Router->dispatch(…)	~/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php	214
    17	Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(…)		
    16	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	141
    15	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)	~/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php	55
    14	Illuminate\View\Middleware\ShareErrorsFromSession->handle(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	125
    13	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)	~/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php	61
    12	Illuminate\Session\Middleware\StartSession->handle(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	125
    11	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)	~/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php	36
    10	Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	125
    9	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)	~/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php	40
    8	Illuminate\Cookie\Middleware\EncryptCookies->handle(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	125
    7	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)	~/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php	42
    6	Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	125
    5	Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(…)		
    4	call_user_func(…)	~/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php	101
    3	Illuminate\Pipeline\Pipeline->then(…)	~/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php	115
    2	Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(…)	~/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php	84
    1	Illuminate\Foundation\Http\Kernel->handle(…)	~/index.php	44